addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/pscan/WebSocketPassiveScannerDecorator.java

```diff
@@ -0,0 +1,52 @@
+package org.zaproxy.zap.extension.websocket.pscan;
+
+import org.zaproxy.zap.extension.websocket.WebSocketMessageDTO;
+
+abstract class WebSocketPassiveScannerDecorator implements  WebSocketPassiveScanner{
+	protected final WebSocketPassiveScanner webSocketPassiveScanner;
+	
+	public WebSocketPassiveScannerDecorator(WebSocketPassiveScanner webSocketPassiveScanner){
+		this.webSocketPassiveScanner = webSocketPassiveScanner;
+	}
+	
+	@Override
+	public String getName() {
+		return webSocketPassiveScanner.getName();
+	}
+	
+	@Override
+	public void scanMessage(WebSocketPassiveScanThread thread, WebSocketMessageDTO webSocketMessage) {
+		webSocketPassiveScanner.scanMessage(thread, webSocketMessage);
+	}
+	
+	@Override
+	public int getId() {
+		return webSocketPassiveScanner.getId();
+	}
+	
+	@Override
+	public int hashCode() {
+		return webSocketPassiveScanner.getId();
+	}
+	
+	@Override
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
```

Comment:
Shouldn't the null check be before the equality check? Otherwise in the previous if statement the obj could be null and NPE when checking equality.